src/Mappers/InventoryMapper.php

```diff
@@ -133,97 +151,120 @@ public function createInventoryDTOsFromVariation($variationData, $itemMappingMet
 
     $nextAvailableDate = $this->getAvailableDate($mainVariationId); // Pending. Need Item
 
-    // the 'stock' element is not declared for the Variation type,
-    // so type hints for "variationData" need to stay as "array"
-    $stockList = $variationData['stock'];
-    /** @var VariationStock $stock */
-    foreach ($stockList as $stock) {
-      $warehouseId = $stock['warehouseId'];
-      if (!isset($warehouseId)) {
-        // we don't know the warehouse, so we can't figure out the supplier ID.
-        // Not an error, but unexpected.
-        $loggerContract->info(
-          TranslationHelper::getLoggerKey(self::LOG_KEY_STOCK_MISSING_WAREHOUSE),
-          [
-            'additionalInfo' => [
-              'variationID' => $mainVariationId,
-              'variationNumber' => $variationNumber
-            ],
-            'method' => __METHOD__
-          ]
-        );
-        continue;
+    $filters = [self::STOCK_COL_VARIATION_ID => $mainVariationId];
+    if (isset($w3cStart) && !empty($w3cStart)) {
+      // FIXME: these conversions should happen before calling into here
+      $filters[self::STOCK_FILTER_UPDATED_AT_FROM] = $w3cStart;
+
+      if (isset($w3CEnd) && !empty($w3CEnd)) {
+        $filters[self::STOCK_FILTER_UPDATED_AT_TO] = $w3CEnd;
       }
+    }
 
-      // this is a 'mixed' value - might be null.
-      $supplierId = $this->getSupplierIDForWarehouseID($warehouseId);
+    /**
+     * @var StockRepositoryContract
+     */
+    $stockRepository = pluginApp(StockRepositoryContract::class);
+    $stockRepository->setFilters($filters);
 
-      if (!isset($supplierId) || $supplierId === 0 || $supplierId === '0') {
-        // no supplier assigned to this warehouse - NOT an error - we should NOT sync it
-        $loggerContract->debug(
-          TranslationHelper::getLoggerKey(self::LOG_KEY_NO_SUPPLIER_ID_ASSIGNED_TO_WAREHOUSE),
-          [
-            'additionalInfo' => ['warehouseID' => $warehouseId, 'variationID' => $mainVariationId,],
-            'method' => __METHOD__
-          ]
-        );
-        continue;
-      }
+    $pageNumber = 1;
+    do {
+      $stockSearchResponsePage = $stockRepository->listStock(InventoryMapper::ALL_STOCK_COLS, $pageNumber, 50);
```

Comment:
This is one of the biggest changes we are making.
We incorrectly used to search for "All Variations that changed since last time we sync'd" which only picked up changes in the metadata for products, and did not filter on changes in **stock**, which is what we wanted.

So, now we find "All Variations that represent Products to be sold on Wayfair," and we use a Stock Repository to find the Stock that meets our "changed since" criteria.